recipes/mailio/all/patches/adapt-cmakelists.patch

```diff
@@ -0,0 +1,60 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 9cd343d..b7cc91c 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,8 +1,6 @@
+ cmake_minimum_required (VERSION 3.16.3)
+ 
+ project(mailio VERSION 0.20.0)
+-set(CMAKE_CXX_STANDARD 17)
```

Comment:
Why do you need removing c++17 standard, but at same time, forcing it on cmake definitions?
I understand it can support multiple standards, but usually, projects support a main standard version only.

Reply:
I think if user wants to be able to build everything using C++20, for example, he should be able to do that.

Reply:
but are there C++20 features on this project? The upstream agrees with that?

Reply:
There are no C++20 features.

If that's ok to always build with C++17, then I will delete this small change and add a warning if the cppstd is set and not C++17.

Reply:
No problem keeping the hardcoded CMAKE_CXX_STANDARD in CMakeLists.txt. When you pass `-s compiler.cppstd=20`, Conan will pass `-DCONAN_CMAKE_CXX_STANDARD="20" -DCONAN_STD_CXX_FLAG="-std=c++2a"` which will override the hardcoded version.

I made a small example here: https://gist.github.com/uilianries/773a47bd0eb150f1c1cce3f9b95d53a0